driver-core/src/main/java/com/datastax/driver/core/SessionManager.java

```diff
@@ -377,7 +378,11 @@ public void onFailure(Throwable t) {
                         } else {
                             logger.warn("Error creating pool to " + host, t);
                         }
-                        future.set(false);
+                        if (t instanceof Error) {
```

Comment:
This is not strictly related to JAVA-1431, but we agreed that errors should be propagated.